pkg/operator/target_status_test.go

```diff
@@ -92,29 +90,35 @@ func expandPopulateTargetsTestCases(testCases []populateTargetsTestCase) []popul
 				},
 				Status: copy.Status,
 			}
-			for endpointIndex, endpointStatus := range clusterPodMonitoring.Status.EndpointStatuses {
-				clusterPodMonitoring.Status.EndpointStatuses[endpointIndex].Name = podMonitoringScrapePoolToClusterPodMonitoringScrapePool(endpointStatus.Name)
+			for idx, status := range cpm.Status.EndpointStatuses {
+				cpm.Status.EndpointStatuses[idx].Name = podMonitoringScrapePoolToClusterPodMonitoringScrapePool(status.Name)
 			}
-			clusterPodMonitorings = append(clusterPodMonitorings, clusterPodMonitoring)
+			clusterPodMonitorings = append(clusterPodMonitorings, cpm)
+		}
+		dataPodMonitorings := updateTargetStatusTestCase{
+			desc:           tc.desc + "-pod-monitoring",
+			targets:        tc.targets,
+			podMonitorings: tc.podMonitorings,
+			expErr:         tc.expErr,
 		}
-		dataPodMonitorings := populateTargetsTestCase{
-			name:              data.name + "-pod-monitoring",
-			prometheusTargets: data.prometheusTargets,
-			podMonitorings:    data.podMonitorings,
+		dataFinal = append(dataFinal, dataPodMonitorings)
+		if len(clusterPodMonitorings) == 0 {
```

Comment:
It looks like this was originally checking the length of `podMonitorings` and skipping the test, but it looks like you moved this down here because you want to append at least one for the kubelet tests. Does it make sense for the kublet tests to be suffixed with "-pod-monitoring" when there is none? In other words, why don't we move the check back up, change it back to check for `podMonitoring` and if there are none, then we could add the test case maybe without a suffix or with a `-no-pod-monitoring` suffix?

Reply:
Yea - that's cleaner than what I did. Done.